packages/react-reconciler/src/ReactFiberRoot.js

```diff
@@ -151,3 +161,51 @@ export function createFiberRoot(
 
   return root;
 }
+
+export function isRootSuspendedAtTime(
+  root: FiberRoot,
+  expirationTime: ExpirationTime,
+): boolean {
+  const firstSuspendedTime = root.firstSuspendedTime;
+  const lastSuspendedTime = root.lastSuspendedTime;
+  return (
+    firstSuspendedTime !== NoWork &&
+    (firstSuspendedTime >= expirationTime &&
+      lastSuspendedTime <= expirationTime)
+  );
+}
+
+export function markRootSuspendedAtTime(
+  root: FiberRoot,
+  expirationTime: ExpirationTime,
+): void {
+  const firstSuspendedTime = root.firstSuspendedTime;
+  const lastSuspendedTime = root.lastSuspendedTime;
+  if (firstSuspendedTime < expirationTime) {
```

Comment:
These branches would be easier to read with the comparison flipped (compare `expirationTime` against the current values)

Reply:
Oh, right, ExpirationTime is a decreasing clock

Reply:
ExpirationTime using to be increasing but we flipped it because it tends to result in fewer comparisons to `NoWork`: https://github.com/facebook/react/pull/13912